File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 1
    q = deque(priorities)
    index = location
    while len(q)>1 :
        pop = q.popleft()
        if pop < max(q):
            q.append(pop)
            if index == 0:
                index = len(q) - 1
            else:
                index -= 1
        else:
            if index == 0:
                return answer
            else:
                answer += 1
                index -= 1
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 1;
    
    let queue = priorities;
    let index = location;
    
    while(queue.length) {
        let pop = queue.shift();
        
        // pop가 작으면 다시 큐에 삽입
        if(pop < Math.max(...queue)) {
            queue.push(pop);
            if(index === 0) {
                index = queue.length - 1; 
            }
            else {
                index -= 1;
            }
        }
        // pop가 크거나 같으면 실행
        else {
            if(index === 0){
                return answer;
            }
            else {
                answer += 1;
                index -= 1;
            }
        }
    }

    return answer;
}